jobrunner/lib/database.py

```diff
@@ -153,37 +164,104 @@ def get_connection():
     filename = config.DATABASE_FILE
     # Looks icky but is documented `threading.local` usage
     cache = CONNECTION_CACHE.__dict__
+    if filename not in cache:
+        raise Exception("Database not initialised")
+    return cache[filename]
+
+
+class MigrationNeeded(Exception):
+    pass
+
+
+def init_db(filename=None, migrations=None):
```

Comment:
Related to the comment below, it looks like `init_db()` bundles up both opening the connection and creating the database file if it doesn't exist. But I thought we wanted to separate those?